exporters/otlp/otlp.go

```diff
@@ -178,8 +182,8 @@ func (e *Exporter) dialToCollector() (*grpc.ClientConn, error) {
 	}
 
 	ctx := context.Background()
-	if len(e.c.headers) > 0 {
-		ctx = metadata.NewOutgoingContext(ctx, metadata.New(e.c.headers))
+	if e.metadata.Len() > 0 {
```

Comment:
This pattern of checking the length and then annotating the context is repeated a few times now. Probably a good idea to unify this into a method.